### PR TITLE
drivers: nrf*: drop -pin properties support

### PR DESCRIPTION
**Pull request by gmarull**
This patch series drop support for the deprecated `-pin` properties. Zephyr 3.4, the next release, will remove pinmux as well as legacy pin configuration mechanisms.

Ref https://github.com/zephyrproject-rtos/zephyr/pull/55025

### CODE REVIEW

**Review by mbolivar-nordic — COMMENTED** [review]
I think it makes sense to keep pinctrl_nrf_migrate for one more release.

Remember that many people won't take action on deprecations until it is absolutely necessary. Therefore, I consider it likely that we will see some subset of users only finding out about this script and wanting to run it once their `-pin` based DTS files no longer work.

Keeping the script around for one more release is thus useful for them since they will still have it in the tree that they want to test against. It can then be removed in v3.5.

Basically I am saying I think it makes sense to keep the migration mechanism around in the first release that makes migration mandatory.

**Comment by mbolivar-nordic** [comment]
(otherwise LGTM)

**Comment by gmarull** [comment]
> I think it makes sense to keep pinctrl_nrf_migrate for one more release.
> 
> Remember that many people won't take action on deprecations until it is absolutely necessary. Therefore, I consider it likely that we will see some subset of users only finding out about this script and wanting to run it once their `-pin` based DTS files no longer work.
> 
> Keeping the script around for one more release is thus useful for them since they will still have it in the tree that they want to test against. It can then be removed in v3.5.
> 
> Basically I am saying I think it makes sense to keep the migration mechanism around in the first release that makes migration mandatory.

Removed patch dropping migration tool. Will mention it again on the release notes when we reach 3.4-rc1.

**Review by mbolivar-nordic — DISMISSED** [review]

File: soc/arm/nordic_nrf/common/soc_nrf_common.h [hunk 1]
```diff
@@ -180,61 +180,16 @@
...
-		DT_NODE_PATH(node_id) " has legacy *-pin properties"	\
-			" defined although PINCTRL is enabled");	\
-	BUILD_ASSERT(IS_ENABLED(CONFIG_PINCTRL) ||			\
-		     !DT_NUM_PINCTRL_STATES(node_id),			\
-		DT_NODE_PATH(node_id) " has pinctrl states defined"	\
-			" although PINCTRL is not enabled")
+#define NRF_DT_CHECK_NODE_HAS_PINCTRL_SLEEP(node_id)			       \
+	BUILD_ASSERT(!IS_ENABLED(CONFIG_PM_DEVICE) ||			       \
+		     (IS_ENABLED(CONFIG_PM_DEVICE) &&			       \
+		      DT_PINCTRL_HAS_IDX(node_id, 1)),			       \
```
[the remark this answers is not in the document]
Comment (gmarull) [reply soc/arm/nordic_nrf/common/soc_nrf_common.h]: done, also used `DT_PINCTRL_HAS_NAME(node_id, sleep)`

File: drivers/spi/spi_nrfx_spi.c [hunk 2]
```diff
@@ -393,34 +385,8 @@ static int spi_nrfx_init(const struct device *dev)
 #define SPI(idx)			DT_NODELABEL(spi##idx)
 #define SPI_PROP(idx, prop)		DT_PROP(SPI(idx), prop)
 
-#define SPI_NRFX_MISO_PULL(idx)				\
-	(SPI_PROP(idx, miso_pull_up)			\
-		? SPI_PROP(idx, miso_pull_down)		\
```
[the remark this answers is not in the document]
Comment (gmarull) [reply drivers/spi/spi_nrfx_spi.c:398]: done

File: drivers/pwm/pwm_nrfx.c [hunk 3]
```diff
@@ -333,40 +323,15 @@ static int pwm_nrfx_pm_action(const struct device *dev,
...
-			(PWM_CH_INVERTED(dev_idx, ch_idx)		\
-			 ? NRFX_PWM_PIN_INVERTED : 0)),			\
-		(NRFX_PWM_PIN_NOT_USED))
-
 #define PWM_NRFX_DEVICE(idx)						      \
-	NRF_DT_CHECK_PIN_ASSIGNMENTS(PWM(idx), 1,			      \
-				     ch0_pin, ch1_pin, ch2_pin, ch3_pin);     \
-	static struct pwm_nrfx_data pwm_nrfx_##idx##_data = {		      \
-		COND_CODE_1(CONFIG_PINCTRL, (),				      \
-			(.initially_inverted =				      \
```
[the remark this answers is not in the document]
Comment (gmarull) [reply drivers/pwm/pwm_nrfx.c:351]: thanks, used local variable

File: drivers/spi/spi_nrfx_spi.c (hunk 2 above)
[the remark this answers is not in the document]
Comment (anangl) [reply drivers/spi/spi_nrfx_spi.c:398]: not quite
https://github.com/zephyrproject-rtos/zephyr/blob/c5b1b85b7afec30f9061725b1e9108b5ced6537b/dts/bindings/spi/nordic%2Cnrf-spi.yaml#L10-L11

[the remark this answers is not in the document]
Comment (gmarull) [reply drivers/spi/spi_nrfx_spi.c:398]: ops, forgot non spim

**Review by anangl — APPROVED** [review]

**Review by mbolivar-nordic — APPROVED** [review]
